smalld/smalld.py

```diff
@@ -108,25 +118,55 @@ def run(self):
 
             self.gateway = Gateway(gateway_url)
 
-            for data in self.gateway:
-                logger.debug("gateway payload received: %s", data)
-                for listener in self.listeners:
-                    listener(AttrDict(json.loads(data)))
+            try:
+                for data in self.gateway:
+                    logger.debug("gateway payload received: %s", data)
+                    for listener in self.listeners:
+                        listener(data)
+            except GatewayClosedException as e:
+                if e.code in recoverable_error_codes:
+                    logger.error(f"gateway closed: {e}")
```

Comment:
We don't want to log this as an error. We can expect the gateway to close and reconnections to happen.